hazelcast/src/main/java/com/hazelcast/internal/management/ManagementCenterService.java

```diff
@@ -130,6 +130,10 @@ public ManagementCenterService(HazelcastInstanceImpl instance) {
         }
     }
 
+    protected TimedMemberStateFactory createTimedMemberStateFactory(HazelcastInstanceImpl instance) {
```

Comment:
I guess this method is some kind of template method that can be overridden the EnterprisemanagementCenter?

If so; I think having a policy based approach is way cleaner.